emcc.py

```diff
@@ -2534,6 +2535,16 @@ def phase_final_emitting(options, target, wasm_target, memfile):
     shared.JS.handle_license(final_js)
     shared.run_process([shared.PYTHON, shared.path_from_root('tools', 'hacky_postprocess_around_closure_limitations.py'), final_js])
 
+  # Unmangle previously mangled `import.meta` references in both main code and libraries.
+  # See also: `preprocess` in parseTools.js.
+  if settings.EXPORT_ES6 and settings.USE_ES6_IMPORT_META:
+    with open(final_js, 'r+') as f:
+      src = f.read()
+      src = src.replace('EMSCRIPTEN$IMPORT$META', 'import.meta')
+      f.seek(0)
+      f.write(src)
+      f.truncate()
```

Comment:
Almost everything else we do to the JS is not directly in-place like this but instead creates a new temporary file which allows a debug build to see the complete history of the JS file as it processed (the sets final_js to point to the new file).  Maybe this is overkill, and it would be nice to create better, more centralized way of doing this.. but maybe better to stick with the existing convention here?

Reply:
SGTM, I don't mind. I'm not entirely sure I understand the mechanism - aside from renaming files, do I need to also call `save_intermediate`? What does it do?

Reply:
That seems to be the pattern so far yes... as I say it would be nice to refactor/encapsulate that at some point.